### DISCUSSION TITLE
the middleware combined for Next-Auth and Next-Intl doesn't work in production.

### DISCUSSION THREAD

**Discussion by 4Furki4** [amannn/next-intl#442] [Q&A]
Hello, I followed [this guide ](https://next-intl-docs.vercel.app/docs/routing/middleware#example-auth-js) for combining next-auth and next-intl. It works on the local dev and production environment. However, the live version deployed by Vercel doesn't work.

Here's what I diagnosed:

- After signing in to the application, I'm sure that Next-Auth works because it signs the user in and shows the protected content in the public page.
- Next-Intl works as expected as well.

What is the problem ? 

I have private pages such as '/my-learning', '/saved', and '/profile'. These pages are protected before signing in, the app redirects users to the '/sign-in' page and can be shown after signing in with credentials or providers in both local dev env and prod build.

In [the live version](https://learn-u-team-7-git-develop-learnu-team-7.vercel.app/), even though users successfully sign in, these pages stay protected but the profile icon in the sidebar and the content in the homepage appears.

May you please help me to solve this problem ? I'm really frustrated :(

**Comment by amannn** [comment]
Hmm, the deployed version of the guide you're mentioning is here: https://example-next-13-next-auth-next-intl.vercel.app

After you login with admin/admin, you can visit the `/secret` page.

Maybe you can find something when comparing your app with the example?

**Reply by 4Furki4** [reply]
There was just one difference I have in my code which is the custom session callback function in NextAuth configs. I changed it but didn't do the trick. Couldn't see any further difference so I'll be using manual redirection using session hook and conditions since the deadline is near. 

Thanks for helping me

**Comment by Albosonic** [comment]
I am having the same issue. Everything works great locally, but when deployed on Vercel; 
Once my app does a redirect to a protected page after being authenticated successfully. The Middleware kicks me out back to the homepage(a non protected page).
I'm baffled as to why it works great locally but not when deployed on Vercel.

Here is my repo: https://github.com/Albosonic/Yachty

Any help is appreciated.
